src/GraphQl/Resolver/Factory/ResolverFactory.php

```diff
@@ -35,7 +35,7 @@ public function __invoke(?string $resourceClass = null, ?string $rootClass = nul
             // Data already fetched and normalized (field or nested resource)
             if ($body = $source[$info->fieldName] ?? null) {
                 // special treatment for nested resources without a resolver/provider
-                if ($operation instanceof Query && $operation->getNested() && !$operation->getResolver() && !$operation->getProvider()) {
+                if ($operation instanceof Query && $operation->getNested() && !$operation->getResolver() && (!$operation->getProvider() || $operation->getProvider() === $resourceClass)) {
```

Comment:
mhh how is this possible? A provider shouldn't be a resource no?

Reply:
Changed it (see comment below)